src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD112AvoidNullReturnInNonAsyncTaskMethodAnalyzer.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Microsoft.VisualStudio.Threading.Analyzers
+{
+    using System.Collections.Immutable;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Operations;
+
+    /// <summary>
+    /// Finds await expressions on <see cref="Task"/> that do not use <see cref="Task.ConfigureAwait(bool)"/>.
+    /// Also works on <see cref="ValueTask"/>.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSTHRD112AvoidNullReturnInNonAsyncTaskMethodAnalyzer : DiagnosticAnalyzer
+    {
+        public const string Id = "VSTHRD112";
+
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            id: Id,
+            title: new LocalizableResourceString(nameof(Strings.VSTHRD112_Title), Strings.ResourceManager, typeof(Strings)),
+            messageFormat: new LocalizableResourceString(nameof(Strings.VSTHRD112_MessageFormat), Strings.ResourceManager, typeof(Strings)),
+            helpLinkUri: Utils.GetHelpLink(Id),
```

Comment:
We'll need a doc for the VSTHRD112 analyzer added to our doc folder (and add a link to it from our index.md file).